src/org/traccar/api/resource/DeviceResource.java

```diff
@@ -122,4 +121,34 @@ public Response updateTotalDistance(DeviceTotalDistance entity) throws SQLExcept
         return Response.noContent().build();
     }
 
+    @Path("/{slave : (geofences|drivers|attributes)}")
```

Comment:
I would suggest use one path for all permissions (`/permissions`), we have all needed information inside request payload.